crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -230,6 +230,7 @@ pub(crate) async fn analyze_ecmascript_module(
             let mut import_references = Vec::new();
 
             let pos = program.span().lo;
+            let is_inside_node_modules = *path.contains("node_modules").await?;
```

Comment:
This seems like a very fragile assertion. What if the path contains `some_folder_that_is_definitely_not_node_modules`? We have a more involved check for "is in directory" in crates/turbopack/src/condition.rs.